### ISSUE TITLE
Display mode causes issues all over the place.

### ISSUE THREAD

**Issue by back2dos** [haxetink/tink_web#62]
Using 4.0.0-preview.4 the following fails to autocomplete:

```haxe
import tink.web.proxy.*;

class Main {
    static function main() {
        var remote = new Remote<Root>(null, null);
        remote.|

    }
}

interface Root {
    @:get
    function now():{now:String};
}
```

**Comment by kevinresol** [comment]
```
[Error - 3:38:07 PM] Request textDocument/completion failed.
  Message: src/Main.hx:6: characters 15-16 : Variables names starting with a dollar are not allowed
Cannot represent { now : String } in JSON because tink.json.Representation<tink.macro.DirectType<9>> cannot be converted to { now : String }
  Code: -32603 
```

**Comment by kevinresol** [comment]
Still not fixed as at Haxe revision cd7ea55
